contracts/test/AllowanceTarget.t.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.7.6;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "contracts/AllowanceTarget.sol";
+import "contracts-test/mocks/MockStrategy.sol";
+import "contracts-test/mocks/MockERC20.sol";
+import "contracts-test/utils/Addresses.sol";
+import "contracts-test/utils/BalanceSnapshot.sol";
+import "contracts-test/utils/StrategySharedSetup.sol";
+
+contract AllowanceTargetTest is Test {
+    using Address for address;
+    using BalanceSnapshot for BalanceSnapshot.Snapshot;
+
+    address newSpender = address(new MockStrategy());
+    address bob = address(0x133701);
+
+    AllowanceTarget allowanceTarget;
+
+    // effectively a "beforeEach" block
+    function setUp() public {
+        // Setup
+        allowanceTarget = new AllowanceTarget(address(this));
+
+        // Label addresses for easier debugging
+        vm.label(address(this), "TestingContract");
+        vm.label(address(allowanceTarget), "AllowanceTarget");
+    }
+
+    /*********************************
+     *          test: setup          *
+     *********************************/
+
+    function testSetupAllowance() public {
+        assertEq(allowanceTarget.spender(), address(this));
+    }
+
+    /*********************************
+     *  test: setSpenderWithTimelock *
+     *********************************/
+
+    function testCantSetSpenderWithTimelockByRandomEOA() public {
```

Comment:
nit: test`Cannot`

Reply:
+1, `Cannot` is clearer than `Cant`

Reply:
Fixed in 0cd05827df343e0d916a159839ea90389e5fc8b1